### PR TITLE
add Azure as a Kubernetes auth provider

### PR DESCRIPTION
**Pull request by goenning**
## Hey, I just made a Pull Request!

This PR adds:

1. Azure Identity as a K8s auth provider. For more: https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview
2. AKS dashboard formatter

This is how the config would look like:

```
        - url: https://my-cluster-aks-53f23f22.privatelink.eastus.azmk8s.io
          name: My AKS Cluster
          authProvider: 'azure'
          skipTLSVerify: true
          dashboardApp: aks #optional
          dashboardParameters: #optional
            subscriptionId: MY-SUB-GUID
            resourceGroup: MY-RG-NAME
            clusterName: my-cluster-aks
```

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [X] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/kubernetes-backend/src/kubernetes-auth-translator/AzureIdentityKubernetesAuthTranslator.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  ): Promise<AzureClusterDetails> {
+    const clusterDetailsWithAuthToken: AzureClusterDetails = Object.assign(
+      {},
+      clusterDetails,
+    );
+
+    const credentials = new DefaultAzureCredential();
+
+    // TODO: can we cache this? It's inneficiant to get a new token every time
+    const accessToken = await credentials.getToken(aksScope);
```
Comment (goenning) [inline plugins/kubernetes-backend/src/kubernetes-auth-translator/AzureIdentityKubernetesAuthTranslator.ts:37]: I'd really like to cache this, but it doesn't seem to be a common practice amongst other auth providers.

The accessToken has a expiryOn property, so we could cache it up to that time and generate a new one after that.

Comment (mclarke47) [reply plugins/kubernetes-backend/src/kubernetes-auth-translator/AzureIdentityKubernetesAuthTranslator.ts:37]: Could we store it as a field? With a setTimeout to refresh? There probably does need to be caching/refresh for server side tokens

**Comment by goenning** [comment]
@mclarke47 do you mind reviewing it when you get a chance? I'm eager to start using it 😁

All the magic happens inside this `DefaultAzureCredential` from Microsoft. When Backstage runs on Azure, it'll have get credentials through Azure. When Backstage runs locally, it'll use developers local token (from $HOME) to authenticate.

File: plugins/kubernetes/src/kubernetes-auth-provider/AzureKubernetesAuthProvider.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ */
+
+import { KubernetesAuthProvider } from './types';
+import { KubernetesRequestBody } from '@backstage/plugin-kubernetes-common';
+
+export class AzureKubernetesAuthProvider implements KubernetesAuthProvider {
+  async decorateRequestBodyForAuth(
+    requestBody: KubernetesRequestBody,
+  ): Promise<KubernetesRequestBody> {
+    // No-op, with aws auth, server's Azure credentials are used for access
```
Comment (freben) [inline plugins/kubernetes/src/kubernetes-auth-provider/AzureKubernetesAuthProvider.ts:24]: is "aws" a typo here?

File: .changeset/big-teachers-dress.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+'@backstage/plugin-kubernetes': patch
+'@backstage/plugin-kubernetes-backend': patch
```
[the remark this answers is not in the document]
Comment (goenning) [reply .changeset/big-teachers-dress.md:3]: Done!

File: plugins/kubernetes/src/kubernetes-auth-provider/AzureKubernetesAuthProvider.ts (hunk 2 above)
Comment (goenning) [reply plugins/kubernetes/src/kubernetes-auth-provider/AzureKubernetesAuthProvider.ts]: copy and paste error :)

**Review by freben — APPROVED** [review]
Looks good!

**Review by mclarke47 — APPROVED** [review]

**Pull request merged**